src/modinfowithconflictinfo.h

```diff
@@ -81,10 +119,32 @@ class ModInfoWithConflictInfo : public ModInfo
 
   bool hasHiddenFiles() const;
 
-private:
+protected:
+
+  /**
+   * @brief Prefetch content for this mod.
+   *
+   * This method can be used to prefetch content from the mod, e.g., for isValid()
+   * or getContents(). This method will only be called when first creating the mod
+   * using multiple threads for all the mods.
+   */
+  virtual void prefetch() override;
 
   // Current game plugin running in MO2:
-  MOBase::IPluginGame const* m_GamePlugin;
+  MOBase::IPluginGame const * const m_GamePlugin;
+
+private:
+
+  /**
+   * @return a file tree for this mod.
+   */
+  std::shared_ptr<const MOBase::IFileTree> updateFileTree() const;
```

Comment:
updateFileTree seems to imply that it will update the internal filetree, maybe generateFileTree or something similar is clearer? It's just the name as the fact that it's const should be pretty clear, but it would make it clearer when just reading the code.